bee-ledger/src/workers/storage.rs

```diff
@@ -0,0 +1,496 @@
+};
+
+use bee_message::{
+    address::{Address, Ed25519Address},
+    milestone::MilestoneIndex,
+    output::{Output, OutputId},
+};
+use bee_storage::{
+    access::{AsStream, Batch, BatchBuilder, Exist, Fetch, Insert, Truncate},
+    backend,
+};
+use bee_tangle::solid_entry_point::SolidEntryPoint;
+
+use std::collections::HashMap;
+
+/// A blanket-implemented helper trait for the storage layer.
+pub trait StorageBackend:
+    backend::StorageBackend
+    + BatchBuilder
+    + Batch<OutputId, CreatedOutput>
+    + Batch<OutputId, ConsumedOutput>
+    + Batch<Unspent, ()>
+    + Batch<(), LedgerIndex>
+    + Batch<MilestoneIndex, OutputDiff>
+    + Batch<(Ed25519Address, OutputId), ()>
+    + Batch<Address, Balance>
+    + Batch<(MilestoneIndex, Receipt), ()>
+    + Batch<(bool, TreasuryOutput), ()>
+    + Exist<Unspent, ()>
+    + Fetch<(), SnapshotInfo>
+    + Fetch<OutputId, CreatedOutput>
+    + Fetch<(), LedgerIndex>
+    + Fetch<Address, Balance>
+    + Fetch<bool, Vec<TreasuryOutput>>
+    + Insert<(), SnapshotInfo>
+    + Insert<(), LedgerIndex>
+    + Insert<(bool, TreasuryOutput), ()>
+    + Truncate<SolidEntryPoint, MilestoneIndex>
+    + for<'a> AsStream<'a, Unspent, ()>
+    + for<'a> AsStream<'a, Address, Balance>
+    + for<'a> AsStream<'a, SolidEntryPoint, MilestoneIndex>
+    + bee_tangle::storage::StorageBackend
+{
+}
+
+impl<T> StorageBackend for T where
+    T: backend::StorageBackend
+        + BatchBuilder
+        + Batch<OutputId, CreatedOutput>
+        + Batch<OutputId, ConsumedOutput>
+        + Batch<Unspent, ()>
+        + Batch<(), LedgerIndex>
+        + Batch<MilestoneIndex, OutputDiff>
+        + Batch<(Ed25519Address, OutputId), ()>
+        + Batch<Address, Balance>
+        + Batch<(MilestoneIndex, Receipt), ()>
+        + Batch<(bool, TreasuryOutput), ()>
+        + Exist<Unspent, ()>
+        + Fetch<(), SnapshotInfo>
+        + Fetch<OutputId, CreatedOutput>
+        + Fetch<(), LedgerIndex>
+        + Fetch<Address, Balance>
+        + Fetch<bool, Vec<TreasuryOutput>>
+        + Insert<(), SnapshotInfo>
+        + Insert<(), LedgerIndex>
+        + Insert<(bool, TreasuryOutput), ()>
+        + Truncate<SolidEntryPoint, MilestoneIndex>
+        + for<'a> AsStream<'a, Unspent, ()>
+        + for<'a> AsStream<'a, Address, Balance>
+        + for<'a> AsStream<'a, SolidEntryPoint, MilestoneIndex>
+        + bee_tangle::storage::StorageBackend
+{
+}
+
+pub(crate) fn insert_output_id_for_address_batch<B: StorageBackend>(
+    storage: &B,
+    batch: &mut <B as BatchBuilder>::Batch,
+    address: &Address,
+    output_id: &OutputId,
+) -> Result<(), Error> {
+    match address {
+        Address::Ed25519(address) => {
+            Batch::<(Ed25519Address, OutputId), ()>::batch_insert(storage, batch, &(*address, *output_id), &())
+                .map_err(|e| Error::Storage(Box::new(e)))?;
+        }
+        address => return Err(Error::UnsupportedAddressKind(address.kind())),
+    }
+
+    Ok(())
```

Comment:
You can remove the `?` on line 95, the `return` on line 97, and the `Ok(())` on line 100.